### PR TITLE
rnbw-254: refactored code for workspaceTreeView

### PR DESCRIPTION
**Pull request by MadDog1994**
I refactored the code in workspaceTreeView,
removed unused variables, moved duplicate code into a variable for further reuse

### CODE REVIEW

**Comment by atulbhatt-system32** [comment]
@MadDog1994 
- Use of custom hook is missing to separate business logic from presentational code.
- Large block of code that are related should be in function.

Check all the necessary instructions mentioned in the description.

**Comment by MadDog1994** [comment]
@atulbhatt-system32 Can you check the changes I made?

File: src/components/main/actionsPanel/workspaceTreeView/helpers/moveActions.ts [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+				type: "error",
+				content: "Folder with the same name already exists.",
+			  });
+			return;
+		  }
+	
+		  // move nested handler-dir to targetHandler with the newName - copy (optional)
+		  try {
+			const newHandler = await targetHandler.getDirectoryHandle(newName, {
+			  create: true,
```
[the remark this answers is not in the document]
Comment (atulbhatt-system32) [reply src/components/main/actionsPanel/workspaceTreeView/helpers/moveActions.ts:48]: convert the block of code inside this try block into a function that you can call inside the try block

File: src/components/main/actionsPanel/workspaceTreeView/helpers/moveActions.ts [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+			// Handle copy (optional)
+			!copy && (await parentHandler.removeEntry(handler.name, { recursive: true }));
+		  } catch (err) {
+			throw new Error("error");
+		  }
+		  
+
+		} else {
+		  // validate if the new name exists
+		  let exists = true;
```
Comment (atulbhatt-system32) [inline src/components/main/actionsPanel/workspaceTreeView/helpers/moveActions.ts:93]: Try to convert the else block also into a separate function with a meaningful name

File: src/components/main/actionsPanel/workspaceTreeView/helpers/moveActions.ts [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+		  // move nested handler-dir to targetHandler with the newName - copy (optional)
+		  try {
+			const dirs = [
+			  {
+				orgPath: nodeData.path,
+				newPath: `${targetNodeData.path}/${newName}`,
+			  },
+			];
+			while (dirs.length) {
+			  const { orgPath, newPath } = dirs.shift() as {
```
Comment (atulbhatt-system32) [inline src/components/main/actionsPanel/workspaceTreeView/helpers/moveActions.ts:162]: Here also apply the same rule

File: src/components/main/actionsPanel/workspaceTreeView/hooks/useFileOperations.ts [hunk 4]
```diff
@@ -0,0 +1,358 @@
...
+		  type: TFileNodeType;
+		}) => {
+		  addRunningActions(["fileTreeView-create"]);
+	
+		  const { parentUid, name, type } = params;
+		  try {
+			// validate
+			const parentNode = ffTree[parentUid];
+			if (parentNode === undefined) throw "error";
+			const parentNodeData = parentNode.data as TFileNodeData;
```
Comment (atulbhatt-system32) [inline src/components/main/actionsPanel/workspaceTreeView/hooks/useFileOperations.ts:47]: Do the modularization of this code inside try block too

File: src/components/main/actionsPanel/workspaceTreeView/hooks/useFileOperations.ts [hunk 5]
```diff
@@ -0,0 +1,358 @@
...
+	
+		  await Promise.all(
+			uids.map(async (uid) => {
+			  try {
+				// validate
+				const node = ffTree[uid];
+				if (node === undefined) throw "error";
+				const nodeData = node.data as TFileNodeData;
+				const parentNode = ffTree[node.parentUid as TNodeUid];
+				if (parentNode === undefined) throw "error";
```
Comment (atulbhatt-system32) [inline src/components/main/actionsPanel/workspaceTreeView/hooks/useFileOperations.ts:100]: Same here

File: src/components/main/actionsPanel/workspaceTreeView/hooks/useNodeActionsHandler.ts [hunk 6]
```diff
@@ -0,0 +1,1275 @@
...
+
+  const { cb_reloadProject } = useReloadProject();
+
+  const { removeInvalidNodes, setInvalidNodes, invalidNodes } = useInvalidNodes();
+
+  const { setTemporaryNodes, removeTemporaryNodes } = useTemporaryNodes();
+
+  const { moveIDBFF, moveLocalFF } = moveActions();
+	
+	const createFFNode = useCallback(
```
Comment (atulbhatt-system32) [inline src/components/main/actionsPanel/workspaceTreeView/hooks/useNodeActionsHandler.ts:82]: Break all the if else blocks with large number of locs into smaller functions and then call them inside those blocks. You can use the concept of pure function if there is only transformation of input to an output is there.

Function names should be proper

File: src/components/main/actionsPanel/workspaceTreeView/hooks/useNodeActionsHandler.ts [hunk 7]
```diff
@@ -0,0 +1,1275 @@
...
+				parentNode.uid
+			  ] as FileSystemDirectoryHandle;
+			if (
+			  !(await verifyFileHandlerPermission(handler)) ||
+			  !(await verifyFileHandlerPermission(parentHandler))
+			) {
+			  addMessage({
+				type: "error",
+				content: `Invalid directory/file. Check if you have "write" permission for the directory/file.`,
+			  });
```
Comment (atulbhatt-system32) [inline src/components/main/actionsPanel/workspaceTreeView/hooks/useNodeActionsHandler.ts:390]: Do the modularization here too

File: src/components/main/actionsPanel/workspaceTreeView/hooks/useNodeActionsHandler.ts [hunk 8]
```diff
@@ -0,0 +1,1275 @@
...
+			  return;
+			}
+		  }
+	
+		  addRunningActions(["fileTreeView-move"]);
+	
+		  const _uids: { uid: TNodeUid; parentUid: TNodeUid; name: string }[] = [];
+		  const _invalidNodes = { ...invalidNodes };
+	
+		  if (project.context === "local") {
```
Comment (atulbhatt-system32) [inline src/components/main/actionsPanel/workspaceTreeView/hooks/useNodeActionsHandler.ts:629]: here also the same

File: src/components/main/actionsPanel/workspaceTreeView/hooks/useNodeActionsHandler.ts [hunk 9]
```diff
@@ -0,0 +1,1275 @@
...
+				  let index = 0;
+				  while (exists) {
+					try {
+					  folderName = `${nodeData.name} copy (${++index})`;
+					  await parentHandler.getDirectoryHandle(folderName, {
+						create: false,
+					  });
+					  exists = true;
+					} catch (err) {
+					  exists = false;
```
Comment (atulbhatt-system32) [inline src/components/main/actionsPanel/workspaceTreeView/hooks/useNodeActionsHandler.ts:1006]: here too

**Comment by MadDog1994** [comment]
@atulbhatt-system32 I took your comments into account and made improvements to the workspaceTreeView

**Review by atulbhatt-system32 — APPROVED** [review]

**Pull request merged**